core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -158,6 +167,44 @@ public String name() {
     return catalogName;
   }
 
+  private boolean shouldSuppressIOException(IOException ioException) {
+    String name = ioException.getClass().getName();
+    return suppressIOException && name.startsWith("org.apache.hadoop.fs.azurebfs");
```

Comment:
I would really prefer if this were more specific and looked at the exception message as well to ensure the error is a permissions or ACL problem.

Reply:
I have fixed the code review suggestions.

I'm detecting the azure blob permission exception using the message (that's all I have available). In that message there is an AzureServiceErrorCode of value "AuthorizationPermissionMismatch" which I use to detect a permission issue.

I agree this is much better error handling since it will let other errors be reported.

I'm thinking the flag should thus be "suppress permission errors". What do you think?

Reply:
That sounds good to me!